### PR TITLE
10351 cftp type error

### PR DESCRIPTION
**Pull request by exarkun**
## Scope and purpose

See https://twistedmatrix.com/trac/ticket/10351

## Contributor Checklist:

* [x] The associated ticket in Trac is here: https://twistedmatrix.com/trac/ticket/10351
* [x] I ran `tox -e lint` to format my patch to meet the [Twisted Coding Standard](https://twistedmatrix.com/documents/current/core/development/policy/coding-standard.html)
* [x] I have created a newsfragment in src/twisted/newsfragments/ (see: [News files](https://twistedmatrix.com/trac/wiki/ReviewProcess#Newsfiles))
* [x] The title of the PR starts with the associated Trac ticket number (without the `#` character).
* [x] I have updated the automated tests and checked that all checks for the PR are green.
* [x] I have submitted the associated Trac ticket for review by adding the word `review` to the keywords field in Trac, and putting a link to this PR in the comment; it shows up in https://twisted.reviews/ now.
* [x] The merge commit will use the below format
  The first line is automatically generated by GitHub based on PR ID and branch name.
  The other lines generated by GitHub should be replaced.

```
Merge pull request #123 from twisted/4356-branch-name-with-trac-id

Author: exarkun
Reviewer: <github_username>, <github_username_if_more_reviewers>
Fixes: ticket:10351

Fix cftp's handling of stderr data from the ssh server.
```

### CODE REVIEW

File: src/twisted/conch/scripts/cftp.py [hunk 1]
```diff
@@ -920,11 +921,44 @@ def _cbSubsystem(self, result):
...
+            # data.
+            #
+            # Currently, only the following type is defined.  Note that the value
+            # for the 'data_type_code' is given in decimal format for readability,
+            # but the values are actually uint32 values.
+            #
+            #             Symbolic name                  data_type_code
+            #             -------------                  --------------
+            #           SSH_EXTENDED_DATA_STDERR               1
+            self.stderr.write(data.decode("utf-8", "backslashreplace"))
```
Comment (moshez) [inline src/twisted/conch/scripts/cftp.py:980]: Why is the choice to use a `backslashreplace` error handler? The standard makes no mention of error handling. Maybe it makes sense to error out by default, and have the developer handle the exception? Maybe pass in a specific error handler explicitly?

This is not a blocker. If it is not changed, maybe it's worthwhile explicitly documenting it as a design choice?

File: src/twisted/conch/test/test_cftp.py [hunk 2]
```diff
@@ -59,16 +60,38 @@ class SSHSessionTests(TestCase):
...
         """
-        stdio = FakeStdio()
-        channel = SSHSession()
-        channel.stdio = stdio
-        channel.eofReceived()
-        self.assertTrue(stdio.writeConnLost)
+        self.channel.eofReceived()
+        self.assertTrue(self.stdio.writeConnLost)
+
+    def test_extReceivedStderr(self) -> None:
```
Comment (moshez) [inline src/twisted/conch/test/test_cftp.py:79]: Would it make sense to add a test with invalid UTF-8 data?

**Review by moshez — APPROVED** [review]
Thank you for working on this! Conch is useful, and it's nice to see some of the rough edges cleaned up.

This seems fine to merge to me. I added a couple of questions and suggestions, which are optional.

File: src/twisted/conch/test/test_cftp.py (hunk 2 above)
Comment (exarkun) [reply src/twisted/conch/test/test_cftp.py:79]: This test covers that with the `\xff` byte it passes.

File: src/twisted/conch/scripts/cftp.py (hunk 1 above)
Comment (exarkun) [reply src/twisted/conch/scripts/cftp.py:980]: I added some more comments about these choices - hopefully they make the motivation for the policy choice more clear (and maybe even guide a future maintainer to making a better choice).

**Pull request merged**